TypeSystem/type-variance/declaration_A01_t05.dart

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion It is a compile-time error if a variance modifier is specified
+/// for a type parameter declared by a static extension, a generic function
+/// type, a generic function or method, or a type alias.
+///
+/// @description Check that it is a compile-time error if a variance modifier is
+/// specified for a type parameter declared by an enum
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=variance
+
+enum  E1<in T> {e;}
```

Comment:
I do not think it should be an error to use statically checked variance here. Where does this rule come from? Perhaps the feature specification should be updated, if it says something that has this error as a consequence (it probably wasn't updated when generalized enums were added).

Reply:
There is no such statement in the spec, I just added it as an experiment and then forgot to delete. Sorry.

Reply:
OK, thanks!